components/org.wso2.carbon.identity.api.server.api.resource/org.wso2.carbon.identity.api.server.api.resource.v1/src/main/java/org/wso2/carbon/identity/api/server/api/resource/v1/util/APIResourceMgtEndpointUtil.java

```diff
@@ -0,0 +1,178 @@
+                        .getRegisteredOIDCScope(ContextLoader.getTenantDomainFromContext());
+                if (registeredOIDCScopes.contains(scope.getName())) {
+                    throw handleException(Response.Status.BAD_REQUEST, ERROR_CODE_RESTRICTED_OIDC_SCOPES);
+                }
+            } catch (IdentityOAuthAdminException e) {
+                throw handleException(Response.Status.INTERNAL_SERVER_ERROR, ERROR_CODE_VALIDATE_SCOPES);
+            }
+        }
+    }
+
+    /**
+     * Validate the attributes provided for search.
+     *
+     * @param attributes List of attributes to be validated.
+     * @return List of validated attributes.
+     */
+    public static List<String> validateAndConvertToLowerCase(List<String> attributes) {
+        List<String> validatedAttributes = new ArrayList<>();
+
+        if (attributes != null && !attributes.isEmpty()) {
+            for (String attribute : attributes) {
+                String lowerCaseAttribute = attribute.toLowerCase(Locale.ENGLISH);
+                if (!APIResourceMgtEndpointConstants.ALLOWED_SEARCH_ATTRIBUTES.contains(lowerCaseAttribute)) {
+                    throw handleException(Response.Status.BAD_REQUEST, ERROR_CODE_INVALID_SEARCH_ATTRIBUTE);
+                } else {
+                    validatedAttributes.add(lowerCaseAttribute);
+                }
+            }
+        }
+        return validatedAttributes;
+    }
+
+    public static APIResourceMgtEndpointException handleException(Response.Status status,
+                                                                  APIResourceMgtEndpointConstants.ErrorMessage error) {
+
+        return new APIResourceMgtEndpointException(status, getError(error.getCode(), error.getMessage(),
+                error.getDescription()));
+    }
+
+    public static APIResourceMgtEndpointException handleException(Response.Status status,
+                                                                  APIResourceMgtEndpointConstants.ErrorMessage error,
+                                                                  String data) {
+
+        return new APIResourceMgtEndpointException(status, getError(error.getCode(), error.getMessage(),
+                String.format(error.getDescription(), data)));
+    }
+
+    public static APIResourceMgtEndpointException handleException(Response.Status status, String errorCode,
+                                                                  String message, String description) {
+
+        return new APIResourceMgtEndpointException(status, getError(errorCode, message, description));
+    }
+
+    public static APIResourceMgtEndpointException handleAPIResourceMgtException(APIResourceMgtException e) {
+
+        Response.Status status = Response.Status.INTERNAL_SERVER_ERROR;
+        if (e instanceof APIResourceMgtClientException) {
+            log.debug(e.getMessage(), e);
```

Comment:
Why we log the error and also throw it, by throwing the error, it automatically log the error

Reply:
In this case, throwing the error doesn't log the error. If the error is a client exception it needs to be logged as a debug log and else (server exception) will be logged as an error log